engines/private/grammar.y

```diff
@@ -0,0 +1,165 @@
+%require "3.6"
+%defines "engines/private/grammar.tab.h"
+%output "engines/private/grammar.tab.cpp"
+
+%{
+
+#include "grammar.h"
+
+#define FORBIDDEN_SYMBOL_ALLOW_ALL
+
+#define	code1(c1)	Private::code(c1);
+#define	code2(c1,c2)	Private::code(c1); Private::code(c2)
+#define	code3(c1,c2,c3)	Private::code(c1); Private::code(c2); Private::code(c3)
+
+#include <stdio.h>
+#include <string.h>
+#include <stdlib.h>
+
+int yydebug=1;
+
+using namespace Private;
+
+extern int yylex();
+extern int yyparse();
+
+void yyerror(const char *str)
+{
+	//fprintf(stderr,"error: %s\n",str);
+}
+
+int yywrap()
+{
+	return 1;
+}
+
+%}
+
+%union {
+	Private::Symbol	*sym;	/* symbol table pointer */
+        int (**inst)();	/* machine instruction */
+        char *s;
+        int *i;
+        int narg;
+}
+
+%token<s> NAME
+%token<sym> STRING NUM
+%type <inst> body if startp cond end expr statements statement fcall value
+%token LTE GTE NEQ EQ FALSETOK TRUETOK NULLTOK IFTOK ELSETOK RECT GOTOTOK DEBUGTOK DEFINETOK SETTINGTOK RANDOMTOK 
+%type<narg> params
+
+%%
+
+lines:   line lines
+       | line
+       ;
+
+line:     DEBUGTOK '{' debug '}'             { /* Not used in the game */ }
+        | DEFINETOK NAME '{' define '}'      { installAll($NAME); }
+        | SETTINGTOK NAME '{' statements '}' { saveSetting($NAME); initSetting(); }
+        ;
+
+debug: /* nothing */
+        | NAME ',' debug
+        ;
+
+statements:  /* nothing */     { $$ = progp; }
+        | statement statements
+
+
+statement: GOTOTOK NAME ';' {
+        $$ = progp;
+	code2(strpush, (Private::Inst) Private::constant(STRING, 0, $NAME));
+        code2(constpush, (Private::Inst) Private::constant(NUM, 1, NULL));
+        code2(strpush, (Private::Inst) Private::constant(STRING, 0, "goto")); 
+        code1(funcpush);
+        }
+        | fcall ';'         { $$ = $1; }   
+        | if cond body end {
+         	/* else-less if */
+		($1)[1] = (Inst)$3;	/* thenpart */
+		($1)[3] = (Inst)$4; 
+                }	/* end, if cond fails */
```

Comment:
Indentation is wrong

Reply:
Removed all tabs, so indentation should be (finally) correct